deps/rabbit/src/rabbit_classic_queue.erl

```diff
@@ -521,59 +542,43 @@ update_msg_status(confirm, Pid, #msg_status{pending = P,
 update_msg_status(down, Pid, #msg_status{pending = P} = S) ->
     S#msg_status{pending = lists:delete(Pid, P)}.
 
+ensure_monitor(_, _, State = stateless) ->
+    State;
+ensure_monitor(Pid, _, State = #?STATE{monitored = Monitored})
+  when is_map_key(Pid, Monitored) ->
+    State;
+ensure_monitor(Pid, QName, State = #?STATE{monitored = Monitored}) ->
+    _ = erlang:monitor(process, Pid, [{tag, {'DOWN', QName}}]),
+    State#?STATE{monitored = Monitored#{Pid => ok}}.
+
 %% part of channel <-> queue api
 confirm_to_sender(Pid, QName, MsgSeqNos) ->
-    %% the stream queue included the queue type refactoring and thus requires
-    %% a different message format
-    case rabbit_queue_type:is_supported() of
-        true ->
-            gen_server:cast(Pid,
-                            {queue_event, QName,
-                             {confirm, MsgSeqNos, self()}});
-        false ->
-            gen_server2:cast(Pid, {confirm, MsgSeqNos, self()})
-    end.
+    Msg = case rabbit_feature_flags:is_enabled(no_queue_name_in_classic_queue_client) of
+              true ->
+                  {confirm, MsgSeqNos, self(), QName};
+              false ->
+                  {confirm, MsgSeqNos, self()}
+          end,
+    gen_server:cast(Pid, {queue_event, QName, Msg}).
 
 send_rejection(Pid, QName, MsgSeqNo) ->
-    case rabbit_queue_type:is_supported() of
-        true ->
-            gen_server:cast(Pid, {queue_event, QName,
-                                  {reject_publish, MsgSeqNo, self()}});
-        false ->
-            gen_server2:cast(Pid, {reject_publish, MsgSeqNo, self()})
-    end.
+    Msg = case rabbit_feature_flags:is_enabled(no_queue_name_in_classic_queue_client) of
```

Comment:
As discussed with @ansd rather than changing the message format we could amend the queue type callbacks involved with the stateful operation to also take the queue name record as an argument. This way we don't need to maintain the extra queue name (which uses memory for known but obscurely technical reasons with how maps work) in the queue type state (as it is used in the queue type state map as the key)

Reply:
I removed the unnecessary feature flag and all queue names in the queue clients.